flax/nnx/summary.py

```diff
@@ -40,6 +40,15 @@
 except ImportError:
   in_ipython = False
 
+# Ensure None is dumped as the string 'None' (and not YAML 'null') for clarity
```

Comment:
Unfortunately, this will affect any other use of the yaml library as well. If `flax` users try to dump some yaml with this library loaded, they'd see Nones instead of nulls unexpectedly. We have to subclass the SafeDumper to keep these changes localized:

```python
class NoneDumper(yaml.SafeDumper):
  pass

NoneDumper.add_representer(
    type(None),
    lambda dumper, data: dumper.represent_scalar('tag:yaml.org,2002:str', 'None')
  )
```

Then, instead of calling `yaml.safe_dump`, we'd need to call the more generic `yaml.dump_all` with the `Dumper=NoneDumper` kwarg.